common/param.go

```diff
@@ -0,0 +1,565 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package common
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/viper"
+	structpb "google.golang.org/protobuf/types/known/structpb"
+)
+
+func (c *ParamDescription) Apply(value interface{}) error {
+	valueKind := reflect.ValueOf(value).Kind()
+	kind := reflect.Kind(c.Kind)
+
+	if valueKind != kind {
+		if kind == reflect.Int || kind == reflect.Int32 || kind == reflect.Int64 ||
+			kind == reflect.Int16 || kind == reflect.Int8 {
+			switch t := value.(type) {
+			case float64:
+				if t != float64(int(t)) {
+					return fmt.Errorf("expected type %q, but found %q", kind.String(), valueKind.String())
+				}
+			default:
+				return fmt.Errorf("expected type %q, but found %q", kind.String(), valueKind.String())
+			}
+		} else if kind == reflect.Uint || kind == reflect.Uint32 || kind == reflect.Uint64 ||
+			kind == reflect.Uint16 || kind == reflect.Uint8 {
+			switch t := value.(type) {
+			case float64:
+				if t != float64(int(t)) || t < 0 {
+					return fmt.Errorf("expected type %q, but found %q", kind.String(), valueKind.String())
+				}
+			default:
+				return fmt.Errorf("expected type %q, but found %q", kind.String(), valueKind.String())
+			}
+
+		} else {
+			return fmt.Errorf("expected type %q, but found %q", kind.String(), valueKind.String())
+		}
+	}
+
+	return nil
+}
+
+func NewParamStore(name string) *ParamStore {
+	store := new(ParamStore)
+	err := store.Init(name)
+	if err != nil {
+		return nil
+	}
+	return store
+}
+
+func (s *ParamStore) Init(name string) error {
+	data, err := structpb.NewStruct(map[string]interface{}{})
+	if err != nil {
+		return err
+	}
+
+	s.Name = name
+	s.Data = data
+	s.Params = make(map[string]*ParamDescription)
+	s.IsFrozen = false
+
+	return nil
+}
+
+func (s *ParamStore) DefineParam(name string, kind reflect.Kind, path string, necessity ParamNecessity) error {
```

Comment:
For this method, it would benefit, if you could please add a short description of what the method is doing!